templates/repo/diff/comments.tmpl

```diff
@@ -5,7 +5,7 @@
 	{{if .OriginalAuthor}}
 		<span class="avatar"><img src="{{AppSubUrl}}/assets/img/avatar_default.png"></span>
 	{{else}}
-		{{template "shared/user/avatarlink" Dict "Context" $.Context "user" .Poster}}
+		{{template "shared/user/avatarlink" Dict "Context" $.root.Context "user" .Poster}}
```

Comment:
What's the difference between `Dict "Context" $.root.Context` and `Dict "ctx" $.root`? I  can see `Dict "ctx" $.root` below, which is using `$.root` as the ctx directly.

Reply:
`Dict "Context" $.root.Context` allows us to use `$.Context` in the underlying template.
`Dict "ctx" $.root` allows us to use `$.ctx` in the underlying template.

This is a data map, `Dict` could help to compose them into the data map.

Reply:
I know that basic knowledge.

I mean What's the difference between `Dict "Context" $.root.Context` and `Dict "ctx" $.root`?

Why not use `Dict "Context" $.root` since the `$.root` is also a context used widely in this template?

Reply:
if we use `Dict "Context" $.root`, then we need `$.Context.Context` in the underlying template.

Reply:
OK, I think I know the problem now. 

`$.root` is a map (not a Context interface), `Dict "ctx" $.root` is passing the map as name `ctx`, it's not a real context.

The naming is quite misleading .......